Automattic/woocommerce-payments
# Add Capture Charge button on payment details page

Fixes 

#### Changes proposed in this Pull Request

* Add a button on Payment Details page to capture an authorized payment that is yet to be finalized.

A glimpse at the capture flow: (pending final changes; see TODO below)  

https://user-images.githubusercontent.com/1288616/188133926-de5b1309-ef35-4243-b6cc-5d21adfaea1e.mp4

#### Testing instructions

* Go to WP Admin → Payments → Settings and make sure the checkbox "Issue an authorization on checkout, and capture later" is checked.
* Create a new order using WC frontend. (eg. buy a Beanie using a [test card](https://woocommerce.com/document/payments/testing/#test-card-numbers))
* Open the order details page from WP Admin → WooCommerce → Orders.
* Click the payment intent link to visit the order's payment details page.
* Verify that you see a notice "You need to capture this charge before" along with a capture button.
* Click the Capture button.
* Verify that the payment is successfully captured. (the notice disappears, a success message appears, timeline is refreshed)
* Refresh the page and verify that the capture notice doesn't appear.

#### TODO

- [x] Notice background color as per design.
- [x] Remove mockings.
- [x] Get authorization id from query params.
- [x] Streamline `submitCaptureAuthorization` to support both "Uncaptured transactions" and "Payment details" pages.

-------------------


**Post merge**
